WIP: First draft of specific Language WG process. (#20688)
PowerShell/PowerShell: docs/community/working-group.md

   It's worth noting that Working Group members who repeatedly speak on behalf of the Working Group without
   consensus, they may be censured or removed from the team.

> 1. The behavior of the Language Working Group is slightly modified from the work flow as show below.

Reviewer: Seems like the number shouldn't be here and this should be a separate paragraph

There may be situations where a working group member might exhibit behavior that is objectionable,
exceed the authority defined in this document or violate the [Code of Conduct](../../CODE_OF_CONDUCT.md).
We recommend to report such issues by using the [Report Content](https://docs.github.com/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam) mechanism to bring it to the attention of the maintainers to review.

## Language Working Group Process

The PowerShell Language WG will follow the [C#](https://github.com/dotnet/csharplang) model for language enhancements.

> For _new feature proposals_, please create a [discussion](https://github.com/orgs/PowerShell/discussions/new?category=enhancements),
> and only submit a pull request if invited to do so by a member of the Language WG (a "champion").

Reviewer 2: ```suggestion
For _new feature proposals_, please create a [discussion](https://github.com/orgs/PowerShell/discussions/new?category=enhancements).
A member of the Language WG may choose to "champion" a proposal, which means
they will bring it to the Language WG meeting. Pull requests for language features
will only be considered when the author has been invited by the Language WG to
open the pull request.
```

### Discussions

> Debate pertaining to language features takes place in the form of [Discussions](https://github.com/PowerShell/discussions) in this repo.

Reviewer 2: ```suggestion
Debate pertaining to language features takes place in the form of [Discussions](https://github.com/orgs/PowerShell/discussions) in this repo.
```

If you want to suggest a feature, discuss current design notes or proposals, etc.,

> please [open a new Discussion topic](https://github.com/orgs/PowerShell/discussions/new?category=enhancements),

Reviewer 2: ```suggestion
please [open a new Discussion topic](https://github.com/orgs/PowerShell/discussions/new?category=enhancements).
```

minor punctuation nit

> Discussions that are short and stay on topic are much more likely to be read.
> If you leave comment number fifty, chances are that only a few people will read it.

Reviewer 2: ```suggestion
```

While absolutely true, I'm not sure including it here will have a positive impact

To make discussions easier to navigate and benefit from, please observe a few rules of thumb:

> - Discussion should be relevant to PowerShell language design. If they are not, they will be summarily closed.

Reviewer 2: ```suggestion
- Focus on the impact the change would have, ideally with real world use cases. Language changes must meet a very high bar for impact to be considered due to their cost (in implementation and maintenance)
- Discussion should be relevant to PowerShell language design. If they are not, they will be summarily closed.
```

- Choose a descriptive topic that clearly communicates the scope of discussion.
- Stick to the topic of the discussion. If a comment is tangential, or goes into detail on a subtopic, start a new discussion and link back.

> - Is your comment useful for others to read, or can it be adequately expressed with an emoji reaction to an existing comment?

Reviewer 2: ```suggestion
- Consider expressing support for a proposal with a reaction emoji rather than a comment, if the comment would otherwise not add more relevant information
```

> Proposals that only make existing syntax optionally illegal will be rejected by the language WG to prevent increased language complexity.

Reviewer 2: ```suggestion
Proposals that introduce an additional dialect are very unlikely to be considered without incredibly large and broad impact due to the significant increase in complexity that accompanies them.
```

### Proposals

Once you have a fully fleshed out proposal describing a new language feature in syntactic and semantic detail,

> please [open an issue for it](https://github.com/powershell/powershell/issues/new/choose),

Reviewer 2: I think it might make more sense to leave those as discussions and require the Language WG to create an issue if a member chooses to champion a proposal. Otherwise I think we'll just end up with a discussion and an issue for every proposal, rather than just an issue.

and it will be labeled as a [Proposal](https://github.com/PowerShell/PowerShell/issues?q=is%3Aopen+is%3Aissue+label%3AProposal).
The comment thread on the issue can be used to hash out or briefly discuss details of the proposal,
as well as pros and cons of adopting it into PowerShell.
If an issue does not meet the bar of being a full proposal, we may move it to a discussion, so that it can be "baked" further.
Specific open issues or more expansive discussion with a proposal will often warrant opening a side discussion
rather than cluttering the comment section on the issue.

When a member of the Language WG finds that a proposal merits discussion,
they can [Champion](https://github.com/PowerShell/PowerShell/issues?q=is%3Aopen+is%3Aissue+label%3A%22Champion%22) it,
which means that they will bring it to the Language WG meeting.

> If the WG decides to work on adopting the feature, the proposer,

Reviewer 2: ```suggestion
If the WG decides to work on adopting the feature then the proposer,
```

I think it reads a little easier but feel free to ignore ofc
